src/ai/backend/manager/models/gql_models/container_registry_utils.py

```diff
@@ -0,0 +1,251 @@
+            "reference": "project",
+            "reference_id": harbor_project_id,
+        })
+
+        async with sess.get(get_quota_id_api, allow_redirects=False, **rqst_args) as resp:
+            if resp.status != 200:
+                raise InternalServerError(f"Failed to get quota info! response: {resp}")
+
+            res = await resp.json()
+            if not res:
+                raise ObjectNotFound(object_name="quota entity")
+            if len(res) > 1:
+                raise InternalServerError(
+                    f"Multiple quota entities found. (project_id: {harbor_project_id})"
+                )
+
+            previous_quota = res[0]["hard"]["storage"]
+            quota_id = res[0]["id"]
+
+            return HarborQuotaInfo(previous_quota=previous_quota, quota_id=quota_id)
+
+    async def read(self) -> int:
+        if not self.registry.is_global:
+            get_assoc_query = sa.select(
+                sa.exists()
+                .where(AssociationContainerRegistriesGroupsRow.registry_id == self.registry.id)
+                .where(AssociationContainerRegistriesGroupsRow.group_id == self.group_row.row_id)
+            )
+            assoc_exist = (await self.db_sess.execute(get_assoc_query)).scalar()
+
+            if not assoc_exist:
+                raise ValueError("The group is not associated with the container registry.")
+
+        ssl_verify = self.registry.ssl_verify
+        connector = aiohttp.TCPConnector(ssl=ssl_verify)
+        async with aiohttp.ClientSession(connector=connector) as sess:
+            rqst_args: dict[str, Any] = {}
+            rqst_args["auth"] = aiohttp.BasicAuth(
+                self.registry.username,
+                self.registry.password,
+            )
+
+            previous_quota = (await self._get_quota_info(sess, rqst_args))["previous_quota"]
+            if previous_quota == -1:
+                raise ObjectNotFound(object_name="quota entity")
+
+            return previous_quota
+
+    async def create(self, quota: int) -> None:
+        ssl_verify = self.registry.ssl_verify
+        connector = aiohttp.TCPConnector(ssl=ssl_verify)
+        async with aiohttp.ClientSession(connector=connector) as sess:
+            rqst_args: dict[str, Any] = {}
+            rqst_args["auth"] = aiohttp.BasicAuth(
+                self.registry.username,
+                self.registry.password,
+            )
+
+            quota_info = await self._get_quota_info(sess, rqst_args)
+            previous_quota, quota_id = quota_info["previous_quota"], quota_info["quota_id"]
+
+            if previous_quota > 0:
+                raise GenericBadRequest(f"Quota limit already exists. (gr: {self.project_id})")
+
+            put_quota_api = yarl.URL(self.registry.url) / "api" / "v2.0" / "quotas" / str(quota_id)
+            payload = {"hard": {"storage": quota}}
+
+            async with sess.put(
+                put_quota_api, json=payload, allow_redirects=False, **rqst_args
```

Comment:
Do we need to use the PUT method for the create operation?

Reply:
Yes, put request is required.
This has already been discussed in Teams, but for detailed information, refer to the Harbor API documentation.